packages/core-utils/src/app/contract-deployment.ts

```diff
@@ -108,3 +117,28 @@ export const deploy = async (
 
   return deployContractFunction(wallet)
 }
+
+/**
+ * Returns an address's first deployed contract or `null` if that address hasn't
+ * created a contract yet. This is useful when creating a contract when a service
+ * is initialized and using that contract on subsequent runs.
+ * @param addresss The address that is being deployed from
+ * @returns contractAddress The address of the first deployed contract or `null` if one hasn't been deployed yet 
+ */
+export const getFirstDeployedContractAddress = async (
+  provider: Provider,
+  address: string,
+): Promise<string | null> => {
+  const nonce = 0
+
+  const contractAddress = getAddress(add0x(keccak256(RLP.encode([
```

Comment:
Can use core-utils utils/crypto.ts `keccak256(...)` to centralize dependencies where possible.